arbos/programs/api.go

```diff
@@ -264,21 +260,30 @@ func newApiClosures(
 		balance := evm.StateDB.GetBalance(address)
 		return common.BigToHash(balance), cost
 	}
-	accountCode := func(address common.Address) ([]byte, uint64) {
-		cost := vm.WasmAccountTouchCost(evm.StateDB, address)
-		if !evm.StateDB.Empty(address) {
-			return evm.StateDB.GetCode(address), cost
+	accountCode := func(address common.Address, gas uint64) ([]byte, uint64) {
+		// In the future it'll be possible to know the size of a contract before loading it.
+		// For now, require the worst case before doing the load.
+
+		metaCost := vm.WasmAccountTouchCost(evm.StateDB, address)
+		loadCost := 3 * am.WordsForBytes(params.MaxCodeSize)
+		sentry := am.SaturatingUAdd(metaCost, loadCost)
```

Comment:
Is a saturating add really what is needed here? Wouldn't it actually be a problem if this overflowed?